packages/common/test/image_loaders/image_loader_spec.ts

```diff
@@ -0,0 +1,154 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {CommonModule} from '@angular/common';
+import {NgOptimizedImageModule} from '@angular/common/src/directives/ng_optimized_image';
+import {RuntimeErrorCode} from '@angular/common/src/errors';
+import {IMAGE_LOADER, ImageLoader, ImageLoaderConfig} from '@angular/common/src/image_loaders/image_loader';
+import {ImgixLoader} from '@angular/common/src/image_loaders/imgix_loader';
+import {Component} from '@angular/core';
+import {ComponentFixture, TestBed} from '@angular/core/testing';
+import {expect} from '@angular/platform-browser/testing/src/matchers';
+
+describe('Built-in image directive loaders', () => {
+  describe('Imgix loader', () => {
+    describe('invalid paths', () => {
+      it('should throw if path is empty', () => {
+        expect(() => {
+          setupTestingModule([{provide: IMAGE_LOADER, useValue: ImgixLoader('')}]);
+        })
+            .toThrowError(
+                `NG0${RuntimeErrorCode.INVALID_INPUT}: ImgixLoader has detected an invalid path: ` +
+                `expecting a path like http://somepath.imgix.net/` +
+                `but got: \`\``);
+      });
+
+      it('should throw if not a path', () => {
+        expect(() => {
+          setupTestingModule([{provide: IMAGE_LOADER, useValue: ImgixLoader('wellhellothere')}]);
```

Comment:
I think we should consider packaging loaders into functions (that would return a provider), something like this:
```suggestion
          setupTestingModule([provideImgixLoader('wellhellothere')]);
```
This would be inline with other functions that we plan to expose as public APIs in the future. In this case image loader functions would look consistent, for example:
```
const appRef = bootstrapApplication(RootCmp, {
  providers: [
    provideImgixLoader('wellhellothere'),
    provideRouter( ... ),
  ]
});
```
// cc @pkozlowski-opensource

Reply:
Yep, this wouldn't hurt.